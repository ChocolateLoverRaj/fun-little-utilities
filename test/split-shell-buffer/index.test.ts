import Splitter from 'split-shell-buffer'

const normalText = [
  'abc def ghi',
  'jkl mno pqrs'
].join('\n')

const styledText = [
  'abc \x1B[1mdef \x1B[2m',
  'ghi \x1B[0;0mjkl \x1B[91;22;3m',
  'mno pqrs \x1B[m',
  'tuv wxyz',
  '01 \x1B[2m23 \x1B[1m45',
  '67 \x1B[m89',
  'foo',
  'bar'
].join('\n')

it('correctly indents normal text', () => {
  const indentedNormalText = [
    '  abc def ghi',
    '  jkl mno pqrs'
  ].join('\n')

  expect(
    Splitter
      .fromString(normalText)
      .withIndent(2)
      .toString()
  ).toBe(
    Splitter
      .fromString(indentedNormalText)
      .toString()
  )
})

it('correctly indents styled text', () => {
  expect(
    Splitter
      .fromString(styledText)
      .withIndent(2)
      .toString()
  ).toMatchSnapshot()
})
