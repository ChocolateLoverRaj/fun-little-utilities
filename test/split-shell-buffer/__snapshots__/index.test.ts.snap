// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly indents styled text 1`] = `
"  abc [1mdef [2m
[0m  [1m[2mghi [0;0mjkl [91;22;3m
[0m  [91;22;3mmno pqrs [m
  tuv wxyz
  01 [2m23 [1m45
[0m  [2m[1m67 [m89
  foo
  bar"
`;
